main.cpp

```diff
@@ -32,24 +32,52 @@ int main(int argc, char* argv[])
     cu::fs::path input_struc_path;
     cu::fs::path input_mat_path;
     cu::fs::path output_struc_path;
+    std::vector<double> rotation_axis_input;
+    double rotation_angle_input=0;
+
+
 
     app.add_option("-s,--structure",input_struc_path,"Path to input structure (vasp format).")->required()->check(CLI::ExistingFile);
-    app.add_option("-m,--matrix",input_mat_path,"Path to file containing 3x3 transformation matrix.")->required()->check(CLI::ExistingFile);
+    app.add_option("-m,--matrix",input_mat_path,"Path to file containing 3x3 transformation matrix.")->check(CLI::ExistingFile);
+    //add conditional requiring -a if -r (from CLI readme)
+    auto* rotation_opt = app.add_option("-a,--rotation-axis",rotation_axis_input,"Rotation axis in vector form")->expected(3);
+    app.add_option("-r,--rotation-angle",rotation_angle_input,"Rotation angle in radians");
     app.add_option("-o,--output",output_struc_path,"Path to write transformed structure to.")->required();
 
     CLI11_PARSE(app, argc, argv);
 
     // Read a structre
     auto input_struc=cu::xtal::Structure::from_poscar(input_struc_path);
 
+    if(rotation_opt->count()) {
+    //Read rotation inputs and normalize the axis
+    double j = rotation_axis_input[0];
```

Comment:
You can make the intent more clear with a loop:
```
Eigen::Vector3d rotation_axis;
for(int i=0; i<3; ++i)
{
    rotation_axis(i)=rotation_axis_input[i];
}
```
If you want to be extra fancy you can change `rotation_axis_input` to be of type `std::array<double,3>`. Then use structured bindings:
```
auto [j,k,l] = rotation_axis_input;
Eigen::Vector3d rotation_axis(j,k,l);
```